### DISCUSSION TITLE
Does helmfile run helm concurrently?

### DISCUSSION THREAD

**Discussion by canon-cmre-craig-matear** [helmfile/helmfile#1691] [Questions]
We are seeing lots of error like this intermittently 
Error: error loading /helm/.cache/helm/repository/istio-index.yaml: empty index.yaml file

Since its intermittent I'm guessing its being caused by this:
https://github.com/helm/helm/issues/10735

Anybody else seeing this, and know of a workaround? I tried replacing the helm executable with a script that forces the calling process to wait if there is already a helm process running. It seems to fix the issue, but its really slow.

**Comment by yxxhero** [comment]
@CraigMatear we will try to find an perfect solution.

**Reply by canon-cmre-craig-matear** [reply]
@yxxhero great, thanks for looking into this. Do you need anything from me?

**Comment by canon-cmre-craig-matear** [comment]
Hey in-case this is for finding a solution in helmfile (or for somebody else with the same problem)

replacing the helm binary with a script like this 

```
#!/usr/bin/env bash 
args=""
if [ "$1" == "repo" ]; then 
args="-e" 
fi 
( 
flock $args --no-fork 200
helm_orig $@) 200>/tmp/helm_lockfile
```

Seems to be fixing the issue. It adds a little bit of time to the execution time.

**Reply by zeemy23** [reply]
this appears to work with a slight change

```
#!/usr/bin/env bash
args=""
if [ "$1" == "repo" ]; then
    args="-e"
fi

# Create lock directory if it doesn't exist
mkdir -p /tmp/helm-locks

# Use a proper redirection to set up the file descriptor
exec 200>/tmp/helm-locks/helm_lockfile
flock $args 200
# Execute the original command
helm_orig "$@"
# Release the lock
exec 200>&-
```
